src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -474,15 +487,29 @@ private void validateIntegerPartitioningColumn(String columnName, FailureCollect
     }
   }
 
-  private void validateTimePartitioningColumn(String columnName, FailureCollector collector, Schema fieldSchema) {
+  private void validateTimePartitioningColumn(String columnName, FailureCollector collector,
+                                              Schema fieldSchema, TimePartitioning.Type timePartitioningType) {
+
     Schema.LogicalType logicalType = fieldSchema.getLogicalType();
-    if (logicalType != LogicalType.DATE && logicalType != LogicalType.TIMESTAMP_MICROS
-      && logicalType != LogicalType.TIMESTAMP_MILLIS) {
+
+    boolean isNotTimestamp = logicalType != LogicalType.TIMESTAMP_MICROS && logicalType != LogicalType.TIMESTAMP_MILLIS;
```

Comment:
Nit: Can we use isTimestampField instead of isNotTimestamp to avoid multiple negations which are harder to read

 boolean isTimestampField = (logicalType == LogicalType.TIMESTAMP_MICROS || logicalType == LogicalType.TIMESTAMP_MILLIS);

boolean isDateField = logicalType == LogicalType.DATE;

Reply:
Yes, this is more readable. 
Made changes